c/tskit/trees.c

```diff
@@ -3638,9 +3637,10 @@ D_prime_summary_func(tsk_size_t state_dim, const double *state,
         double p_B = p_AB + p_aB;
 
         double D = p_AB - (p_A * p_B);
-        if (D >= 0) {
```

Comment:
The purpose of this change is to avoid dividing by zero. When D is zero, the denominator becomes zero.

Reply:
Should the result be NaN then? We should probably mark this as a follow-up issue, these things can be tricky to nail down.

Reply:
That's a good question. Should D' (and other ratio statistics like r and r^2) return zero if there is no variation, or NaN? I had thought those results should be zero at first, but now I'm not so sure... We could tackle this in a follow-up issue, like you say.